.github/workflows/go.yml

```diff
@@ -192,7 +192,7 @@ jobs:
             retries=0
 
             while [ $retries -lt $max_num_retries ]; do
-              if $PYROSCOPE_PATH exec --apiKey=${{ secrets.PYROSCOPE_CLOUD_TOKEN }} -- gotestsum --rerun-fails=5 --packages="$pkg" --format standard-verbose -- -v -coverpkg="$pkg" -coverprofile=profile.cov $pkg; then
+              if $PYROSCOPE_PATH exec --apiKey=${{ secrets.PYROSCOPE_CLOUD_TOKEN }} -- gotestsum --rerun-fails=3 --packages="$pkg" --format standard-verbose -- -v -coverpkg="$pkg" -coverprofile=profile.cov $pkg; then
```

Comment:
@dwasse are we sure this isn't 3*3=9 retries? Seems like 3 per package a total of 3 times: https://github.com/gotestyourself/gotestsum#re-running-failed-tests

If that's intended behavior/we don't know how to avoid we should document this

Reply:
Yeah the intention is allow retrying all agents tests as well as individual packages. Perhaps we can lower even more? Really just trying to find the appropriate setting that will allow broken tests to fail without having to wait a long time.